packages/dd-trace/test/ci-visibility/exporters/agentless/exporter.spec.js

```diff
@@ -58,4 +66,40 @@ describe('CI Visibility Exporter', () => {
       expect(writer.flush).to.have.been.called
     })
   })
+
+  describe('when ITR is enabled', () => {
+    it('should append a code coverage payload when exportCoverage is called', () => {
+      const testSpan = {
+        context: () => ({ _traceId: '1', _spanId: '2' })
+      }
+      const payload = { testSpan, coverageFiles: ['file.js'] }
+
+      exporter = new Exporter({ url, flushInterval: 0, isIntelligentTestRunnerEnabled: true })
+
+      exporter.exportCoverage(payload)
+      expect(coverageWriter.append).to.have.been.calledWith({
+        traceId: '1',
+        spanId: '2',
+        files: ['file.js']
+      })
+      expect(coverageWriter.flush).to.have.been.called
+    })
+    it('should flush after the configured flush interval', function (done) {
+      this.timeout(5000)
+      exporter = new Exporter({ url, flushInterval, isIntelligentTestRunnerEnabled: true })
+
+      const testSpan = {
+        context: () => ({ _traceId: '1', _spanId: '2' })
+      }
+      const payload = { testSpan, coverageFiles: ['file.js'] }
+
+      exporter.exportCoverage(payload)
+
+      setTimeout(() => {
+        expect(coverageWriter.flush).to.have.been.called
+        done()
+      }, flushInterval)
+      expect(coverageWriter.append).to.have.been.called
```

Comment:
Could use a check to make sure it's not called immediately.

```suggestion
      expect(coverageWriter.append).to.have.been.called
      expect(coverageWriter.flush).to.not.have.been.called
```